### ISSUE TITLE
Make use of tree-sitter.injection-regex to match language to be injected

### ISSUE THREAD

**Issue by stsewd** [nvim-treesitter/nvim-treesitter#812]
I realized a code block wasn't being highlighted as javascript, this is because the capured name was "js", but looks like the parser name is being used to match the injected language https://github.com/nvim-treesitter/nvim-treesitter/blob/6aaf240d0c6ad10676e252f2e89110820d706718/lua/nvim-treesitter/parsers.lua#L32

tree-sitter declares some fields in the package.json file that holds some information on how to match an injected language, for js for example

https://github.com/tree-sitter/tree-sitter-javascript/blob/3f8b62f9befd3cb3b4cb0de22f6595a0aadf76ca/package.json#L32

Not sure if it's too complicated to use that same file/field, but another option would be to just declare that regex in `parsers.lua`.
Haven't looked at the api for language injection, but I think that's being done in the neovim side.

**Comment by vigoux** [comment]
This have been done on neovim's side, and we handle that with a custom `injections.scm` query though

**Comment by stsewd** [comment]
@vigoux yeah I know the injections query, but a fenced block in markdow or rst can have `js` as file type, and it doesn't get highlighted because it expects to have "javascript" instead. The injection regex is meant to solve that problem.

**Comment by kyazdani42** [comment]
the injections.scm file is supposed to match a node and set it's filetype. If `js` is set it's probably an error in some injections.scm query ?

**Comment by stsewd** [comment]
@kyazdani42 markdown and rst have a field that defines the language

``````
```{language}
```
``````

there language can be anything, that field is captured with `@language`

**Comment by kyazdani42** [comment]
i see, maybe we could just defined a mapping from 'js' to 'javascript' then, i don't think it would be too hard (like extending the lang_to_ft dict)

**Comment by steelsojka** [comment]
> i see, maybe we could just defined a mapping from 'js' to 'javascript' then, i don't think it would be too hard (like extending the lang_to_ft dict)

I think this would be tricky since the injections are performed in neovim and not in nvim-treesitter.
